src/gap_to_julia.jl

```diff
@@ -78,164 +94,153 @@ function gap_to_julia(::Type{Rational{T}}, x::Int64) where T <: Integer
     len = Globals.Length( obj )
-    result = BitArray( undef, len )
+    result = BitArray(undef, len)
     for i in 1:len
       result[i] = obj[i]
     end
     return result
 end
 
-## Arrays
-function gap_to_julia( ::Type{Array{Obj,1}}, obj :: GapObj, recursion_dict = IdDict() )
-    if ! Globals.IsList( obj )
-        throw(ArgumentError("<obj> is not a list"))
-    end
-    if haskey(recursion_dict,obj)
-        return recursion_dict[obj]
-    end
-    len_list = length(obj)
-    new_array = Array{Any,1}( undef, len_list)
-    recursion_dict[obj] = new_array
-    for i in 1:len_list
-        new_array[i] = ElmList(obj, i)  # returns 'nothing' for holes in the list
-    end
-    return new_array
-end
-
-function gap_to_julia( ::Type{Array{T,1}}, obj :: GapObj, recursion_dict = IdDict() ) where T
-    if ! Globals.IsList( obj )
-        throw(ArgumentError("<obj> is not a list"))
-    end
-    if haskey(recursion_dict,obj)
-        return recursion_dict[obj]
-    end
-    len_list = length(obj)
-    new_array = Array{T,1}( undef, len_list)
-    recursion_dict[obj] = new_array
-    for i in 1:len_list
-        current_obj = ElmList(obj, i)
-        new_array[i] = get!(recursion_dict, current_obj) do
-            gap_to_julia(T, current_obj, recursion_dict)
+## Vectors
+function gap_to_julia(::Type{Vector{T}}, obj::GapObj, recursion_dict=IdDict(); recursive=true) where T
+    ! Globals.IsList(obj) && throw(ConversionError(obj, Vector{T}))
+    if ! haskey(recursion_dict, obj)
+        len_list = length(obj)
+        new_array = Vector{T}(undef, len_list)
+        recursion_dict[obj] = new_array
+        for i in 1:len_list
+            current_obj = ElmList(obj, i)  # returns 'nothing' for holes in the list
+            if recursive
+                new_array[i] = get!(recursion_dict, current_obj) do
+                    gap_to_julia(T, current_obj, recursion_dict;
+                                 recursive=true)
+                end
+            else
+                new_array[i] = current_obj
+            end
         end
     end
-    return new_array
+    return recursion_dict[obj]
 end
 
-## Matrix / list-of-lists
-function gap_to_julia( ::Type{Array{T,2}}, obj :: GapObj, recursion_dict = IdDict() ) where T
-    if ! Globals.IsList( obj )
-        throw(ArgumentError("<obj> is not a list"))
-    end
-    if haskey(recursion_dict, obj)
+## Matrices or lists of lists
+function gap_to_julia(::Type{Matrix{T}}, obj::GapObj, recursion_dict=IdDict(); recursive=true) where T
+    if Globals.IsMatrixObj(obj)
+        if ! haskey(recursion_dict, obj)
+            elm = Globals.MatElm
+            nrows = Globals.NumberRows(obj)
+            ncols = Globals.NumberColumns(obj)
+            new_array = Matrix{T}(undef, nrows, ncols)
+            recursion_dict[obj] = new_array
+            for i in 1:nrows
+                for j in 1:ncols
+                    current_obj = elm(obj, i, j)
```

Comment:
does it return `nothing` for missing entries?

Reply:
or there are no missing entries in GAP Matrices?

Reply:
A GAP matrix has no missing entries